src/kokkos/KokkosCore/atomic.h

```diff
@@ -0,0 +1,102 @@
+#ifndef KokkosCore_atomic_h
+#define KokkosCore_atomic_h
+
+#include <Kokkos_Core.hpp>
+
+// Abstractions to disable atomicity of atomics for serial backend
+namespace cms::kokkos {
+  template <typename T>
+  KOKKOS_FORCEINLINE_FUNCTION T atomic_fetch_add(T* dest, const T val) {
```

Comment:
Note to self that this is fragile at best, and these functions should be (e.g.) templated on the execution space to guarantee different symbols. Probably the `FORCEINLINE` has saved for trouble in the tests so far.

Reply:
For now went with a simpler solution that the bypassing of atomics is an option available when only the Serial backend is enabled. This should be sufficient to demonstrate the impact on performance, after which some better solution can be figured out.